pkg/kubelet/metrics/metrics.go

```diff
@@ -530,6 +555,10 @@ func Register(collectors ...metrics.StableCollector) {
 		legacyregistry.MustRegister(StartedPodsErrorsTotal)
 		legacyregistry.MustRegister(StartedContainersTotal)
 		legacyregistry.MustRegister(StartedContainersErrorsTotal)
+		if runtime.GOOS == "windows" && utilfeature.DefaultFeatureGate.Enabled(features.WindowsHostProcessContainers) {
```

Comment:
So this check is precedented here: https://github.com/kubernetes/kubernetes/blob/ec8e6e877899442da9eb0c87a9612cb4dc20373a/pkg/kubelet/kubelet_pods.go#L153-L155

and here:

https://github.com/kubernetes/kubernetes/blob/ec8e6e877899442da9eb0c87a9612cb4dc20373a/staging/src/k8s.io/kubectl/pkg/util/term/term.go#L93

Some part of me is paranoid that if we gate this on the runtime OS but not the individual metric increments, we could somehow get into a state where we cause a panic trying to increment an unregistered metric.

So I've come around to thinking we should be consistent: if we check `runtime.GOOS == "windows"` at registration, we should also do so at metric increment time. (I still think it should be fine to only check the feature gate at registration time.)

Reply:
> Some part of me is paranoid that if we gate this on the runtime OS but not the individual metric increments, we could somehow get into a state where we cause a panic trying to increment an unregistered metric.

It's okay to increment an unregistered metric, it will not panic.

Reply:
Oh perfect, then this is fine.